custom/enikshay/two_b_release_1/management/commands/enikshay_2b_cbnaat_fix.py

```diff
@@ -0,0 +1,117 @@
+import datetime
+import logging
+
+from corehq.apps.es import CaseSearchES
+from django.core.management import BaseCommand
+from casexml.apps.case.mock import CaseStructure, CaseFactory
+from casexml.apps.case.xform import get_case_updates
+
+logger = logging.getLogger('enikshay_2b_cbnaat_fix')
+
+
+class Command(BaseCommand):
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            'domain',
+            help="The domain to migrate."
+        )
+        parser.add_argument(
+            "log_path",
+            help="Path to write the log to"
+        )
+        parser.add_argument(
+            '--commit',
+            action='store_true',
+            help="actually modifies the cases. Without this flag, it's a dry run."
+        )
+
+    @staticmethod
+    def _get_result_recorded_form(test):
+        """get last form that set result_recorded to yes"""
+        for action in reversed(test.actions):
+            for update in get_case_updates(action.form):
+                if update.id == test.case_id:
+                    case_properties = update.get_update_action().dynamic_properties
+                    if case_properties.get('result_recorded') == 'yes':
+                        return action.form.form_data
+
+    @staticmethod
+    def _get_path(path, form_data):
+        block = form_data
+        while path:
+            block = block.get(path[0], {})
+            path = path[1:]
+        return block
+
+    def handle(self, domain, log_path, **options):
+        commit = options['commit']
+        factory = CaseFactory(domain)
+
+        logger.info("Starting {} migration on {} at {}".format(
+            "real" if commit else "fake", domain, datetime.datetime.utcnow()
+        ))
+
+        cases = (CaseSearchES()
+                 .domain(domain)
+                 .case_type("test")
+                 .case_property_query("test_type_value", "cbnaat", "must")
+                 .scroll())
+
+        with open(log_path, "w") as f:
+            for case in cases:
+                case_props = {prop['key']: prop['value'] for prop in case['case_properties']}
+                updated_by_migration = case_props.get('updated_by_migration')
+                if ((updated_by_migration == 'enikshay_2b_case_properties' or
+                     updated_by_migration == 'enikshay_2b_reason_for_test_fix')
+                        and case.get_case_property('result_recorded') == 'yes'):
+
+                    drug_resistance_list = ''
+                    drug_sensitive_list = ''
+                    resistance_display = None
+
+                    form_data = self._get_result_recorded_form(case)
```

Comment:
This also won't work with the elasticsearch case - it might be easiest to use `CaseAccessors(domain).iter_cases(case_ids)` with the `case_ids` from ES (that'll require a modification to the ES query as well).